### PR TITLE
Amd64/fix some tests

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Fix some tests for x64.

### CODE REVIEW

File: modules/rostests/apitests/dbghelp/testlist.c [hunk 1]
```diff
@@ -8,6 +8,8 @@ extern void func_rsym(void);
 const struct test winetest_testlist[] =
 {
     { "pdb", func_pdb },
+#ifdef _M_IX86
```
Comment (SergeGautherie) [inline modules/rostests/apitests/dbghelp/testlist.c:11]: Maybe use a skip(), as a placeholder/reminder?

File: modules/rostests/winetests/xmllite/reader.c [hunk 2]
```diff
@@ -537,7 +537,7 @@ static void test_reader_create(void)
     IXmlResolver *resolver;
     IUnknown *input, *unk;
     IXmlReader *reader;
-    DtdProcessing dtd;
+    ULONG_PTR dtd; // enum DtdProcessing
```
Comment (HBelusca) [inline modules/rostests/winetests/xmllite/reader.c:540]: Aren't enums going to be always `int`? (especially with MSVC)

File: modules/rostests/winetests/ws2_32/sock.c [hunk 3]
```diff
@@ -7008,6 +7008,11 @@ static void test_write_watch(void)
     size = 0x10000;
     base = VirtualAlloc( 0, size, MEM_RESERVE | MEM_COMMIT | MEM_WRITE_WATCH, PAGE_READWRITE );
     ok( base != NULL, "VirtualAlloc failed %u\n", GetLastError() );
+    if (base == NULL)
+    {
+        skip("failed to allocate %u bytes. Skipping test.", size);
```
Comment (HBelusca) [inline modules/rostests/winetests/ws2_32/sock.c:7013]:
```suggestion
        skip("Failed to allocate %u bytes. Skipping test.", size);
```
How frequently do we get this allocation failing?

File: modules/rostests/apitests/win32nt/ntdd/NtGdiDdQueryDirectDrawObject.c [hunk 4]
```diff
@@ -600,9 +601,6 @@ START_TEST(NtGdiDdQueryDirectDrawObject)
     RTEST(puD3dBufferCallbacks != NULL);
     ASSERT(puD3dDriverData != NULL);
 
-    RTEST(puD3dTextureFormats != NULL);
```
Comment (HBelusca) [inline modules/rostests/apitests/win32nt/ntdd/NtGdiDdQueryDirectDrawObject.c:603]: Why is this removed? Can this pointer be NULL from time to time even e.g. on Windows?

**Review by HBelusca — APPROVED** [review]
Looks globally OK

**Review by GeoB99 — APPROVED** [review]

File: modules/rostests/winetests/xmllite/reader.c (hunk 2 above)
Comment (tkreuzer) [reply modules/rostests/winetests/xmllite/reader.c:540]: In C they are int, yes.

File: modules/rostests/apitests/win32nt/ntdd/NtGdiDdQueryDirectDrawObject.c (hunk 4 above)
Comment (tkreuzer) [reply modules/rostests/apitests/win32nt/ntdd/NtGdiDdQueryDirectDrawObject.c:603]: Please check the code. And then tell me why it wouldn't be NULL.

File: modules/rostests/winetests/ws2_32/sock.c (hunk 3 above)
Comment (tkreuzer) [reply modules/rostests/winetests/ws2_32/sock.c:7013]: It always fails, because we do not support MEM_WRITE_WATCH. I decided to revert the change for now though, because it only makes it worse: instead of crashing, the test hangs later.

File: modules/rostests/winetests/xmllite/reader.c [hunk 5]
```diff
@@ -537,7 +537,11 @@ static void test_reader_create(void)
     IXmlResolver *resolver;
     IUnknown *input, *unk;
     IXmlReader *reader;
+#ifdef __REACTOS__
+    ULONG_PTR dtd;
```
Comment (SergeGautherie) [inline modules/rostests/winetests/xmllite/reader.c:541]:
```suggestion
    LONG_PTR dtd;
```

File: modules/rostests/winetests/xmllite/reader.c [hunk 6]
```diff
@@ -603,7 +607,7 @@ static void test_reader_create(void)
     dtd = 2;
     hr = IXmlReader_GetProperty(reader, XmlReaderProperty_DtdProcessing, (LONG_PTR*)&dtd);
     ok(hr == S_OK, "Expected S_OK, got %08x\n", hr);
-    ok(dtd == DtdProcessing_Prohibit, "got %d\n", dtd);
+    ok(dtd == DtdProcessing_Prohibit, "got %Id\n", dtd);
```
Comment (SergeGautherie) [inline modules/rostests/winetests/xmllite/reader.c:610]: Want an `#ifdef __REACTOS__` too.
Or, better, upstream this fix.

File: modules/rostests/winetests/xmllite/reader.c (hunk 5 above)
Comment (SergeGautherie) [reply modules/rostests/winetests/xmllite/reader.c:541]: Submitted as #4965.

File: modules/rostests/winetests/xmllite/reader.c (hunk 6 above)
Comment (SergeGautherie) [reply modules/rostests/winetests/xmllite/reader.c:610]: Submitted as #4965.
